sql/core/src/main/scala/org/apache/spark/sql/columnar/ColumnStats.scala

```diff
@@ -245,7 +247,8 @@ private[sql] class FixedDecimalColumnStats(precision: Int, scale: Int) extends C
       val value = row.getDecimal(ordinal, precision, scale)
       if (upper == null || value.compareTo(upper) > 0) upper = value
       if (lower == null || value.compareTo(lower) < 0) lower = value
-      sizeInBytes += FIXED_DECIMAL.defaultSize
+      // TODO: this is not right for DecimalType with precision > 18
+      sizeInBytes += 8
```

Comment:
For decimals with precision greater than 18, `CatalystSchemaConverter.minBytesForPrecision` can be useful here. At least it provides a relatively tight upper limit.